heudiconv/main.py

```diff
@@ -0,0 +1,363 @@
+import logging
+import os.path as op
+import sys
+
+from . import __version__, __packagename__
+from .bids import populate_bids_templates, tuneup_bids_json_files
+from .convert import prep_conversion
+from .parser import get_study_sessions
+from .queue import queue_conversion
+from .utils import anonymize_sid, load_heuristic, treat_infofile, SeqInfo
+
+lgr = logging.getLogger(__name__)
+
+
+INIT_MSG = "Running {packname} version {version} latest {latest}".format
+
+
+def is_interactive():
+    """Return True if all in/outs are tty"""
+    # TODO: check on windows if hasattr check would work correctly and add value:
+    return sys.stdin.isatty() and sys.stdout.isatty() and sys.stderr.isatty()
+
+
+def setup_exceptionhook():
+    """
+    Overloads default sys.excepthook with our exceptionhook handler.
+
+    If interactive, our exceptionhook handler will invoke pdb.post_mortem;
+    if not interactive, then invokes default handler.
+    """
+    def _pdb_excepthook(type, value, tb):
+        if is_interactive():
+            import traceback
+            import pdb
+            traceback.print_exception(type, value, tb)
+            # print()
+            pdb.post_mortem(tb)
+        else:
+            lgr.warning(
+              "We cannot setup exception hook since not in interactive mode")
+
+    sys.excepthook = _pdb_excepthook
```

Comment:
FWIW: we have this function not covered by tests -- tested manually seems to still work )